src/mono/mono/mini/mini.c

```diff
@@ -4229,12 +4229,70 @@ mini_get_underlying_type (MonoType *type)
 void
 mini_jit_init (void)
 {
+	mono_os_mutex_init_recursive (&jit_mutex);
+
+#ifndef DISABLE_JIT
 	mono_counters_register ("Discarded method code", MONO_COUNTER_JIT | MONO_COUNTER_INT, &discarded_code);
 	mono_counters_register ("Time spent JITting discarded code", MONO_COUNTER_JIT | MONO_COUNTER_LONG | MONO_COUNTER_TIME, &discarded_jit_time);
 	mono_counters_register ("Try holes memory size", MONO_COUNTER_JIT | MONO_COUNTER_INT, &jinfo_try_holes_size);
 
-	mono_os_mutex_init_recursive (&jit_mutex);
-#ifndef DISABLE_JIT
+	mono_counters_register ("JIT/method_to_ir", MONO_COUNTER_JIT | MONO_COUNTER_LONG | MONO_COUNTER_TIME, &mono_jit_stats.jit_method_to_ir);
```

Comment:
Maybe worth pushing all this into a static function just for readability? Otherwise LGTM.